### ISSUE TITLE
Upgrading from 1.8.1 to 1.13.0 causes exported ONNX file enlarged greatly, printable graph changed, and crashed when converting to TensorRT

### ISSUE THREAD

**Issue by windless0530** [pytorch/pytorch#114677]
# 🐛 Describe the bug

## Project workflow (originally):

1. Train the model on PyTorch v1.8.1

2. Export the checkpoint to ONNX through torch.onnx.export(nn.Module), in the same code repository as the training codes.
```
class _MVFSegmentation(torch.nn.Module):
    def forward(...):
    # ...

def _export_mvf_seg(raw_model, point_features, dump):
    # ...
    torch.onnx.export(
        model,
        tuple(input_.tensor for input_ in inputs),
        f,
        input_names=[input_.name for input_ in inputs],
        output_names=[
            "bbox_pred",
            "cls_pred",
            "dir_pred",
            "segment_pred",
            "anchor_points",
        ],
        verbose=True,
        opset_version=13,
        dynamic_axes={input_.name: input_.dynamic_axes for input_ in inputs},
    )
```

3. Convert the ONNX file to TensorRT, in another code repository, with PyTorch v1.14.0a0+410ce96, TensorRT v8.5.1, CUDA v11.8.
```
def _get_trt_engine(
    onnx_model_path: str,
    input_tensors: List[InputTensor],
    build_cache: bool,
    int8: bool,
) -> bytes:
    # ...
    engine_string = builder.build_serialized_network(network, builder_config)
    # ...
```

4. Do inference with TensorRT runtime.

At first, everything was fine.

## Problem:

Then, I am working on upgrading the PyTorch version of the first 2 steps from v1.8.1 to v1.13.0. Some dependencies are also upgraded, such as CUDA, [mmcv](https://github.com/open-mmlab/mmcv), [mmdet](https://github.com/open-mmlab/mmdetection), etc.

Problem occurs on the 3rd step, converting to TensorRT, where crash (segmentation fault) occurs:
![image](https://github.com/pytorch/pytorch/assets/4645959/8bdb5952-0693-46ed-85ed-e2668f091843)

The call-stack is like:
![image](https://github.com/pytorch/pytorch/assets/4645959/56521f58-6121-4f11-a9e8-e47a5d4ce50c)

## Clues and progress so far:

Here are some clues I've discovered so far.

1. The ONNX file trained and then exported by PyTorch v1.8.1 are sized about 50M, while the v1.13 file sized nearly 500M, almost 10 times.

2. Exporting under v1.13 takes much longer time than v1.8.1.

3. Via observing on [netron.app](https://netron.app/), the ONNX file generated by v1.8.1 contain outputs which seemed to have fixed shape, while the v1.13.0 file seemed to have dynamic shape.
![86v6AnTc7m](https://github.com/pytorch/pytorch/assets/4645959/3ef3b7a0-3ce9-4cbd-a89b-c011183d3d8c)
![xGzqmWMJsw](https://github.com/pytorch/pytorch/assets/4645959/088fed27-086e-459b-a172-1721bd05f494)

4. The node input names format are different.
![vaSZXuPZd2](https://github.com/pytorch/pytorch/assets/4645959/c44e16b5-5bc1-4880-9845-9ca2a40d9f37)
![bFmWtn2GcK](https://github.com/pytorch/pytorch/assets/4645959/8e7a58cf-3f46-47c7-bece-d7fa84b17d7c)

5. Via observing output by `onnx.helper.printable_graph(model.graph)`, the _initializers_ of v1.8.1 ONNX file contain 8 that only appeared as _Constant_ in v1.13.0 ONNX file.
![image](https://github.com/pytorch/pytorch/assets/4645959/cc61b1ce-2be0-4d33-9725-617961a9dd33)

6. Adding `keep_initializers_as_inputs=True` on `torch.onnx.export` does not help remove the above diff (clue No.5).

7. Adding `do_constant_folding=False`, which was default to False in v1.8.1 while True in v.1.13, on `torch.onnx.export`, does not help.

8. v1.8.1 only exported one main graph in ONNX file, while v1.13 would additionally export 8 small sub-graphs.

9. If I train with PyTorch v1.13, and export to onnx under PyTorch v1.8.1, the ONNX file, format, seemes to the same as totally v1.8.1, and no crash when converting to TensorRT.

**So, any suggestion on how I could further detect the root cause and solve it?**

Thank you.

# Versions

Training and exporting:

Ubuntu 20.04
PyTorch 1.8.1 → 1.13.0
CUDA 11.1 → 11.6

Converting to TensorRT and inference:

Ubuntu 20.04
PyTorch 1.14.0a0+410ce96
CUDA 11.8

The most confusing part is, I am making versions of PyTorch and CUDA closer from training to inference procedure, and closer versions caused this issue...

**Comment by windless0530** [comment]
Problem solved by myself.

By switching form TensorRT python SDK to `trtexec` tool, additional information wass printed at crash:

![co6lHLkwwA](https://github.com/pytorch/pytorch/assets/4645959/63b73e00-5862-4d00-acfa-0c91c140d43e)

So I looked up in the onnx graph and found the 4 _if-condition_s are generated by `squeeze` operator. So I replaced `squeeze` with `reshape` to remove the conditional branches and avoided the crash.
